libff/algebra/curves/bls12_381/bls12_381_init.cpp

```diff
@@ -0,0 +1,633 @@
+#include <libff/algebra/curves/bls12_381/bls12_381_g1.hpp>
+#include <libff/algebra/curves/bls12_381/bls12_381_g2.hpp>
+#include <libff/algebra/curves/bls12_381/bls12_381_init.hpp>
+
+namespace libff
+{
+
+bigint<bls12_381_r_limbs> bls12_381_modulus_r;
+bigint<bls12_381_q_limbs> bls12_381_modulus_q;
+
+bls12_381_Fq bls12_381_coeff_b;
+bigint<bls12_381_r_limbs> bls12_381_trace_of_frobenius;
+bls12_381_Fq2 bls12_381_twist;
+bls12_381_Fq2 bls12_381_twist_coeff_b;
+bls12_381_Fq bls12_381_twist_mul_by_b_c0;
+bls12_381_Fq bls12_381_twist_mul_by_b_c1;
+bls12_381_Fq2 bls12_381_twist_mul_by_q_X;
+bls12_381_Fq2 bls12_381_twist_mul_by_q_Y;
+
+bls12_381_Fq bls12_381_g1_endomorphism_beta;
+bigint<bls12_381_r_limbs> bls12_381_g1_safe_subgroup_check_c1;
+
+// Coefficients for G2 untwist-frobenius-twist
+bls12_381_Fq12 bls12_381_g2_untwist_frobenius_twist_v;
+bls12_381_Fq12 bls12_381_g2_untwist_frobenius_twist_w_3;
+bls12_381_Fq12 bls12_381_g2_untwist_frobenius_twist_v_inverse;
+bls12_381_Fq12 bls12_381_g2_untwist_frobenius_twist_w_3_inverse;
+
+bigint<bls12_381_q_limbs> bls12_381_ate_loop_count;
+bool bls12_381_ate_is_loop_count_neg;
+bigint<12 * bls12_381_q_limbs> bls12_381_final_exponent;
+bigint<bls12_381_q_limbs> bls12_381_final_exponent_z;
+bool bls12_381_final_exponent_is_z_neg;
+
+void init_bls12_381_fields()
```

Comment:
IIRC we don't use such function to init the fields, it all goes in `init_<group>_params` (e.g. `init_bls12_377_params`). So the addition of this function is inconsistent.
Let's make sure we are consistent with the other init files and make sure that all variables are initialized here (e.g. to avoid potential issues like the point compression algo not working because the a and b coeffs of the curve aren't init)